sources/PTYSession.m

```diff
@@ -2511,6 +2516,7 @@ - (void)handleKeyPressInCopyMode:(NSEvent *)event {
             [_textview scrollLineNumberRangeIntoView:VT100GridRangeMake(_copyModeState.coord.y, 1)];
         }
         [self.textview setNeedsDisplayOnLine:_copyModeState.coord.y];
+        [self.textview setNeedsDisplayOnColumn:_copyModeState.coord.x];
```

Comment:
Same—only if the vertical guide is enabled

Reply:
Given that this is true any time we call [...textView setNeedsDisplayOnColumn...] I have placed the check there so that most parts of the code don't need to be concerned with this implementation detail (ie the inefficiency of setNeedsDisplayOnCol), but all benefit from this check.